### PR TITLE
[Refactoring] making the code more Succinct  and Pythonic

### PR DESCRIPTION
**Pull request by maldil**
This PR uses the `with` statement and the `.join` function to make the code more Pythonic and Succinct.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the PR.

**Comment by fchollet** [comment]
> Lint / Check the code format (pull_request) Failing after 2m — Check the code format

Please fix code formatting.

File: keras/utils/text_dataset_test.py [hunk 1]
```diff
@@ -74,10 +73,10 @@ def test_text_dataset_from_directory_standalone(self):
         directory = self._prepare_directory(count=7, num_classes=2)
         for i in range(3):
             filename = "text_%s.txt" % (i,)
-            f = open(os.path.join(directory, filename), "w")
-            text = "".join([random.choice(string.printable) for _ in range(20)])
-            f.write(text)
-            f.close()
+            with open(os.path.join(directory, filename), "w") as f:
+                text = "".join([random.choice(string.printable)
+                                for _ in range(20)])
```
Comment (fchollet) [inline keras/utils/text_dataset_test.py:78]: Move line out of the scope

File: keras/utils/text_dataset_test.py [hunk 2]
```diff
@@ -60,12 +60,11 @@ def _prepare_directory(
...
-            f = open(os.path.join(temp_dir, filename), "w")
-            text = "".join(
-                [random.choice(string.printable) for _ in range(length)]
-            )
-            f.write(text)
-            f.close()
+            with open(os.path.join(temp_dir, filename), "w") as f:
+                text = "".join(
+                    [random.choice(string.printable) for _ in range(length)]
+                )
```
Comment (fchollet) [inline keras/utils/text_dataset_test.py:66]: Move line out of the scope

**Review by fchollet — COMMENTED** [review]
Hopefully this might fix the issue.

**Comment by gbaned** [comment]
@maldil Can you please resolve conflicts? Thank you!

File: keras/utils/text_dataset_test.py [hunk 3]
```diff
@@ -56,16 +56,15 @@ def _prepare_directory(
             for path in class_paths:
                 os.mkdir(os.path.join(temp_dir, path))
             paths += class_paths
-
+
```
Comment (fchollet) [inline keras/utils/text_dataset_test.py:59]: Remove

**Comment by fchollet** [comment]
> This branch cannot be rebased due to conflicts

Please resolve this.

**Comment by fchollet** [comment]
> Lint / Check the code format (pull_request) 

Please run `black` as well.

**Comment by fchollet** [comment]
You should push the changes made by black. The code in the PR should pass the black lint check.

**Comment by fchollet** [comment]
> This branch cannot be rebased due to conflicts

Please rebase.

**Comment by fchollet** [comment]
> This branch cannot be rebased due to conflicts

The black check passed! But you still need to rebase.

**Review by fchollet — APPROVED** [review]
Thanks, LGTM

**Pull request merged**